spec/presenters/tree_builder_archived_spec.rb

```diff
@@ -0,0 +1,103 @@
+describe TreeBuilderArchived do
+  context 'TreeBuilderArchived' do
+    before do
+      extend TreeBuilderArchived
```

Comment:
Something seems wrong if we need to create all of these `@` variables in the before block.

Not sure if we are creating too many unnecessary roles, or if there is another way